src/d3d12/nvapi_d3d12_device.cpp

```diff
@@ -0,0 +1,90 @@
+#include "nvapi_d3d12_device.h"
+
+namespace dxvk {
+    Com<ID3D12DeviceExt> GetDevice(ID3D12Device* pDevice, D3D12_VK_EXTENSION extension) {
+        Com<ID3D12DeviceExt> d3d12Device;
+        if (FAILED(pDevice->QueryInterface(IID_PPV_ARGS(&d3d12Device))))
+            return nullptr;
+
+        if (!d3d12Device->GetExtensionSupport(extension))
+            return nullptr;
+
+        return d3d12Device;
+    }
+    
+
+    // We are going to have single map for storing devices with extensions D3D12_VK_NVX_BINARY_IMPORT & D3D12_VK_NVX_IMAGE_VIEW_HANDLE.
+    // These are specific to NVIDIA and both of these extensions goes together.
```

Comment:
@liam-middlebrook  is this a safe assumption?

Reply:
This should be fine, yes.